src/sentry/analytics/event.py

```diff
@@ -9,23 +9,43 @@
 
 
 class Attribute(object):
-    __slots__ = ['name', 'type', 'required']
-
     def __init__(self, name, type=six.text_type, required=True):
         self.name = name
         self.type = type
         self.required = required
 
+    def serialize(self, value):
+        if value is None:
+            return value
+        return self.type(value)
+
 
 class Map(Attribute):
-    def __init__(self, name, attributes, required=True):
+    attributes = ()
+
+    def __init__(self, name, attributes=None, required=True):
         self.name = name
         self.required = required
-        self.attributes = attributes
+        if attributes:
+            self.attributes = attributes
+
+    def serialize(self, value):
+        """
+        If passed a non dictionary we assume we can pull attributes from it.
+
+        This will hard error in some situations if you're passing some obscure
+        value (like an int).
+        """
+        if value is None:
+            return value
 
-    def type(self, value):
         if not isinstance(value, Mapping):
-            raise ValueError('Value must be a dictionary')
+            new_value = {}
```

Comment:
Seems like it'd make sense to make L53 an `else` and just assign this to `items` outright